pyproject.toml

```diff
@@ -84,23 +80,75 @@ mkdocstrings-python = "^1.7.3"
 mkautodoc = "^0.2.0"
 markdown = "^3.5"
 mock = "^5.1.0"
-mypy = "^1.6.1"
-pre-commit = "^3.5.0"
-pylint = "^3.0.2"
-pylint-pydantic = "^0.3.0"
+mypy = "^1.5.1"
 pytest = "^7.4.0"
 pytest-cov = "^4.1.0"
+pytest-ruff = "^0.2.1"
+pytest-mypy = "^0.10.3"
 pytest-mock = "^3.12.0"
-taskipy = "^1.12.0"
-types-mock = "^5.1.0.2"
+poethepoet = "^0.22.0"
+black = "^23.7.0"
+types-mock = "^5.1.0.3"
 types-pyyaml = "^6.0.12"
 types-requests = "^2.31.0"
 types-setuptools = "^68.2.0"
-scikit-learn = "^1.2.1"
-xgboost = "^1.7.4"
+pre-commit = "^2.19.0"
+
+[tool.pytest.ini_options]
+addopts = """
+    --verbose
+    --ruff
+    --mypy
+    --cov-report term
+    --cov-report xml:coverage.xml
+    --cov=src
+    --durations=0
+    --ignore=src/wanna/components/templates
+    --ignore=templates
+    --ignore=samples
+    """
+filterwarnings = [
+    "error",
+    "ignore::UserWarning", # note the use of single quote below to denote "raw" strings in TOML
+    'ignore::DeprecationWarning',
+    'ignore::SyntaxWarning',
+    'ignore::ResourceWarning',
+]
+
+[tool.poe.tasks]
+_black = "black ."
+_ruff = "ruff --select I --fix ."
+format-code = ["_black", "_ruff"]
+docs-deploy = "mkdocs gh-deploy --force"
+docs-serve = "mkdocs serve"
+
+[tool.ruff]
+line-length = 99
```

Comment:
Is there any reason why to use 99? I prefer 120, because that's still quite short, but lets you write longer code than 99 which feels too narrow for todays wide monitors.

Reply:
99 is the default for the CTO projects, I am ok with 120. 
Updated, thx